### PR TITLE
Add a script for building all of the configurations of a benchmark.

### PR DESCRIPTION
**Pull request by davidchisnall**
This takes a base board.json, modifies it to include the different variations, then builds the specified benchmark, once for each version.

Also fixes some bugs for configurations we haven't tested.

### CODE REVIEW

**Comment by davidchisnall** [comment]
@microsoft-github-policy-service agree company="Microsoft"

File: scripts/build_benchmark_configs.sh [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+# Helper script to build multiple configurations of a benchmark.  Patches the
+# board description files to enable and disable different combinations of
+# hardware features.  Also patches include paths to keep track of the original
+# locations.
+
+set -eo pipefail
+
+SCRIPTPATH=$(dirname $(readlink -f $0))
+SDKPATH=$(realpath ${SCRIPTPATH}/../sdk)
```
Comment (nwf-msr) [inline scripts/build_benchmark_configs.sh:11]: Stale?

File: scripts/build_benchmark_configs.sh [hunk 2]
```diff
@@ -0,0 +1,103 @@
+#!/bin/sh
+
+# Helper script to build multiple configurations of a benchmark.  Patches the
+# board description files to enable and disable different combinations of
+# hardware features.  Also patches include paths to keep track of the original
+# locations.
+
+set -eo pipefail
```
Comment (nwf-msr) [inline scripts/build_benchmark_configs.sh:8]: Debian's `dash`, which it uses as `/bin/sh`, apparently doesn't like `-o pipefail`, which seems surprising.

File: scripts/build_benchmark_configs.sh [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+jq 'del(.revoker) | .stack_high_water_mark=false | .defines[.defines| length]|= .+"CHERIOT_FAKE_REVOKER"' < standard.json > ${BOARDNAME}-fake-revoker.json
+rm standard.json
+
+CONFIGS="${BOARDNAME}-software-revoker ${BOARDNAME}-software-revoker-shwm ${BOARDNAME}-hardware-revoker ${BOARDNAME}-hardware-revoker-shwm ${BOARDNAME}-no-revoker ${BOARDNAME}-fake-revoker"
+
+# Patch the include directories so that relative paths become absolute relative
+# to the board file.
+for I in ${CONFIGS}; do
+	# Fix up includes if they expect to be relative to the board file path
+	jq ".driver_includes = (.driver_includes | map_values(. = if . | startswith(\"/\") then . else \"${BOARDPATH}/\" + . end))" < $I.json > $I.json.fixed
```
Comment (nwf-msr) [inline scripts/build_benchmark_configs.sh:82]: I think this also needs to probe for starting with `$`, as in `${sdk}`, which will be expanded by `xmake`?
```suggestion
	jq ".driver_includes = (.driver_includes | map_values(. = if . | startswith(\"/\") or startswith(\"\$\") then . else \"${BOARDPATH}/\" + . end))" < $I.json > $I.json.fixed
```

File: scripts/build_benchmark_configs.sh (hunk 1 above)
Comment (davidchisnall) [reply scripts/build_benchmark_configs.sh:11]: Yes, that’s left over from when I was doing ${sdk} substitution, but I don’t think we need it so I removed the complicated jq expression.

File: scripts/build_benchmark_configs.sh (hunk 3 above)
Comment (davidchisnall) [reply scripts/build_benchmark_configs.sh:82]: No, that will do the wrong thing. If include paths are sdk-relative then we need to do sdk substitution. I had an if then else thing in the jq that substituted ${SDKDIR} for the literal ${sdk} in that case. If you have a file that uses sdk that you can test with, can you try adding that back?

File: scripts/build_benchmark_configs.sh (hunk 2 above)
Comment (davidchisnall) [reply scripts/build_benchmark_configs.sh:8]: Hmm, not sure if it’s better to require bash or skip the pipe fail here. It should be fine with zsh and, I think, the FreeBSD POSIX shell.

File: scripts/build_benchmark_configs.sh (hunk 3 above)
Comment (davidchisnall) [reply scripts/build_benchmark_configs.sh:82]: You need to have it inside the map_values bit.  This line is trying to replace the array, not the elements of the array.

Comment (davidchisnall) [reply scripts/build_benchmark_configs.sh:82]: Yes, that's correct.  I was misremembering why I didn't do this.  We only do ${sdk} substitution for include files and treat relative paths as relative to the board source location.  For the simulator, we also allow ${board}.  That's the thing that needs updating when we move the board file.  Ho hum.

File: scripts/build_benchmark_configs.sh (hunk 2 above)
Comment (nwf-msr) [reply scripts/build_benchmark_configs.sh:8]: Are we actually doing anything in a pipeline in this script?  The only `|`s I see are either part of `||`s or inside `jq` expressions.

Comment (davidchisnall) [reply scripts/build_benchmark_configs.sh:8]: Hmm, maybe not, so just -eo should be fine.

Comment (nwf-msr) [reply scripts/build_benchmark_configs.sh:8]: Just `-e`, I think?  And`-u` is also generally a good plan, but I haven't looked for where we might rely on not-that at the moment.

**Review by nwf-msr — APPROVED** [review]

**Pull request merged**